### PR TITLE
Fixed #29514 -- Made get_default_timezone() use module utc instance.

### PR DESCRIPTION
**Pull request by carltongibson**
Regression in 27ca5ce19f5f184018a61611c1bc319113b1d107.

https://code.djangoproject.com/ticket/29514

### CODE REVIEW

File: django/utils/timezone.py [hunk 1]
```diff
@@ -75,8 +75,9 @@ def get_default_timezone():
     Return the default time zone as a tzinfo instance.
 
     This is the time zone defined by settings.TIME_ZONE.
+    For 'UTC' use module level utc object.
     """
-    return pytz.timezone(settings.TIME_ZONE)
+    return utc if settings.TIME_ZONE.upper() == 'UTC' else pytz.timezone(settings.TIME_ZONE)
```
Comment (carltongibson) [inline django/utils/timezone.py]: Accepts `utc` and `UTC`, but also `uTc` etc. Not sure it's worth trying to do anything more sophisticated here.

**Comment by timgraham** [comment]
@sir-sigurd, any thoughts on this issue?

**Comment by sir-sigurd** [comment]
@timgraham, I think that the same instance should be used for UTC everywhere in Django, but I'm not sure what is the right way to do it. We might add `timezone.get_timezone()` function
```
def get_timezone(name):
    return utc if name.lower() == 'utc' else return pytz.timezone(name)
```
and use it instead of `pytz.timezone()`, but I'm also concerned that return a different type for a one-off case might lead to some issue, although IMHO it doesn't break the contract because `get_default_timezone()` is documented to return "abstract" `tzinfo` instance and we already have some code for handling non-pytz timezones:
https://github.com/django/django/blob/9294110a57ce0a6d14506969c950090045c622c8/django/utils/timezone.py#L260-L269
If we remain unsure, it's OK to revert the original patch.

**Comment by timgraham** [comment]
Okay, I'm inclined to revert the original patch for now. https://github.com/django/django/pull/10104

**Pull request closed without merging**